### PR TITLE
Unwrap script validity field in TxBodyContent

### PR DESCRIPTION
**Pull request by newhoggy**

### CODE REVIEW

**Review by erikd — APPROVED** [review]

**Review by Jimbo4350 — APPROVED** [review]
LGTM! Maybe let Duncan have a look as well.

File: cardano-api/src/Cardano/Api/TxBody.hs [hunk 1]
```diff
@@ -1597,21 +1593,22 @@ makeTransactionBody =
...
 {-# COMPLETE TxBody #-}
 
-getTxBodyContent :: TxBody era -> TxBodyContent ViewTx era
-getTxBodyContent (ByronTxBody body) = getByronTxBodyContent body
-getTxBodyContent (ShelleyTxBody era body _scripts _redeemers mAux _scriptValidity) =
-    fromLedgerTxBody era body mAux
+getTxBodyContent :: TxScriptValidity era -> TxBody era -> TxBodyContent ViewTx era
+getTxBodyContent _ (ByronTxBody body) = getByronTxBodyContent body
+getTxBodyContent scriptValidity (ShelleyTxBody era body _scripts _redeemers mAux _scriptValidity) =
+    fromLedgerTxBody era scriptValidity body mAux
```
Comment (dcoutts) [inline cardano-api/src/Cardano/Api/TxBody.hs:1602]: No we would have to add the validity flag into the `TxBody` type, probably as an extra field in the `ShelleyTxBody` constructor. We have to get the value from the underlying ledger tx type, and not invent it. For one thing we need to do that to have the TxBody round trip test pass, but more importantly we need it to get the right value for the tx from a block.

**Review by dcoutts — DISMISSED** [review]
Good so far, but we have to plumb the flag through to the right place so we can properly have the flag value go back and forth with the underlying ledger tx type (which does contain the flag).

```
type instance Core.Tx (AlonzoEra c) = ValidatedTx (AlonzoEra c)

data ValidatedTx era = ValidatedTx
  { body :: !(Core.TxBody era),
    wits :: !(TxWitness era),
    isValidating :: !IsValidating,
    auxiliaryData :: !(StrictMaybe (Core.AuxiliaryData era))
  }
```
So as you see, the flag value is directly available.

File: cardano-api/src/Cardano/Api/TxBody.hs (hunk 1 above)
Comment (dcoutts) [reply cardano-api/src/Cardano/Api/TxBody.hs:1602]: See also `getAlonzoTxBody` where we extract the parts of the `Alonzo.ValidatedTx`.

Comment (newhoggy) [reply cardano-api/src/Cardano/Api/TxBody.hs]: I remove the argument of `getTxBodyContent` and am taking it from `TxBody`.

**Comment by newhoggy** [comment]
bors merge

**Review by MarcFontaine — APPROVED** [review]
LGTM

**Comment by newhoggy** [comment]
bors merge

**Pull request merged**